pkg/cloudprovider/providers/vsphere/vclib/virtualmachine.go

```diff
@@ -0,0 +1,386 @@
+package vclib
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// VirtualMachine extends the govmomi VirtualMachine object
+type VirtualMachine struct {
+	*object.VirtualMachine
+	datacenter *Datacenter
+}
+
+// IsDiskAttached checks if disk is attached to the VM.
+func (vm *VirtualMachine) IsDiskAttached(ctx context.Context, diskPath string) (bool, error) {
+	// Get object key of controller to which disk is attached
+	_, err := vm.GetVirtualDiskControllerKey(ctx, diskPath)
+	if err != nil {
+		if err == ErrNoDevicesFound {
+			return false, nil
+		}
+		glog.Errorf("Failed to check whether disk is attached to VM: %q. err: %s", vm.Name(), err)
+		return false, err
+	}
+	return true, nil
+}
+
+// GetVirtualDiskUUIDByPath gets the virtual disk UUID by diskPath
+func (vm *VirtualMachine) GetVirtualDiskUUIDByPath(ctx context.Context, diskPath string) (string, error) {
```

Comment:
I have made it consistent with the vdm govmomi functions. Based on the offline discussion, we agreed up on renaming the function to GetVirtualDiskPage83Data as most developer would know it and is already part of VC documentation.